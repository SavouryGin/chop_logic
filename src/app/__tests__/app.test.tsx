import App from 'app';
import React from 'react';
import { render, screen } from '@testing-library/react';

describe('Test App component:', () => {
  it('should display layout components', () => {
    render(<App />);
    expect(screen.getByRole('banner')).toBeInTheDocument();
    expect(screen.getByRole('article')).toBeInTheDocument();
    expect(screen.getByRole('contentinfo')).toBeInTheDocument();
  });

  it('should render the home page by default', () => {
    render(<App />);
    expect(screen.getByRole('article')).toHaveClass('home');
  });

  it('should match the snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
