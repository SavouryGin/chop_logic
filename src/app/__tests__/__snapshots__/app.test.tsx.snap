// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component: should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="layout"
    data-testid="layout"
  >
    <header
      class="header layout__header"
    >
      <div
        class="header__left-panel"
      >
        <button
          class="button button_flat"
          id="button_id_1"
          title="Navigation"
          type="button"
        >
          <span
            class="button__shadow"
          />
          <span
            class="button__edge"
          />
          <span
            class="button__front chop-icon__menu"
            data-testid="button_id_1"
          />
        </button>
        <div
          class="chop-logic-logo"
        >
          <img
            alt="Chop Logic Logo"
            class="country_logo_img"
            role="img"
          />
        </div>
        <h1
          class="header__heading"
        >
          <span
            class="app-link"
            data-testid="app-link"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Chop Logic
            </a>
          </span>
        </h1>
      </div>
      <div
        class="header__right-panel"
      >
        <button
          class="button button_flat"
          id="button_id_2"
          title="Color Theme"
          type="button"
        >
          <span
            class="button__shadow"
          />
          <span
            class="button__edge"
          />
          <span
            class="button__front chop-icon__contrast"
            data-testid="button_id_2"
          />
        </button>
        <button
          class="button button_flat"
          id="button_id_5"
          title="Full Screen"
          type="button"
        >
          <span
            class="button__shadow"
          />
          <span
            class="button__edge"
          />
          <span
            class="button__front chop-icon__enlarge2"
            data-testid="button_id_5"
          />
        </button>
        <button
          class="button button_flat"
          id="button_id_4"
          title="Settings"
          type="button"
        >
          <span
            class="button__shadow"
          />
          <span
            class="button__edge"
          />
          <span
            class="button__front chop-icon__cog"
            data-testid="button_id_4"
          />
        </button>
      </div>
    </header>
    <main
      class="layout__main"
    >
      <article
        class="home"
      />
    </main>
    <footer
      class="footer layout__footer"
    >
      <span
        class="footer__copyright"
      >
        Â© Dmitrii Suroviagin, 2023
      </span>
      <span
        class="footer__links"
      >
        <a
          class="chop-icon__mail4"
          href="mailto:Dmitrii.Suroviagin@gmail.com"
          rel="noreferrer"
          target="_blank"
          title="Mail"
        />
        <a
          class="chop-icon__telegram"
          href="https://telegram.me/savoury_gin"
          rel="noreferrer"
          target="_blank"
          title="Telegram"
        />
        <a
          class="chop-icon__linkedin-square"
          href="https://www.linkedin.com/in/dmitrii-suroviagin/"
          rel="noreferrer"
          target="_blank"
          title="LinkedIn"
        />
        <a
          class="chop-icon__facebook2"
          href="https://www.facebook.com/dmitrii.suroviagin"
          rel="noreferrer"
          target="_blank"
          title="Facebook"
        />
        <a
          class="chop-icon__github"
          href="https://github.com/SavouryGin"
          rel="noreferrer"
          target="_blank"
          title="GitHub"
        />
      </span>
    </footer>
  </div>
</DocumentFragment>
`;
